### PR TITLE
Data Types, Operators and Expressions

### LINKED ISSUE #4: Data Types, Operators and Expressions
N | 问题描述 | 输出格式
----|------|------
1 | 交换2个数x和y，用3+种不同方法来实现 | `{x:d} {y:d} => {:d} {:d}`
2 | 判断某个年份year是否为润年 | `{year:d} => {:d}` 
3 | 判断2个浮点数a和b是否相等 | `{a:f} {b:f} {dt:f} => {:d}`
4 | 判断一个整数n是否为偶数 (even) | `{n:d} => {:d}` 
5 | n为一整数，不用乘除及取模求nx17、n%8、n/16 | `{n:d} => {:d} {:d} {:d}`
6 | 判断一个整数n是否为2<sup>k</sup>(2的k次方) | `{n:d} => {:d} `
7 | 位运算分别实现`chmod 600/+x/u+w/go-w` 4种操作 | `{m:o} => {:o} {:o} {:o} {:o}`

**相应说明**:
1. 同时在PR页面描述各方法的优缺点
1. 输出部为"1/0"，表示“True/False”
1. 输出部为"1/0"，表示“相等/不等”
1. 输出部为"1/0"，表示"even/odd"。要求分别用算术运算符和位运算符实现
1. (无)
1. 输出部为"1/0"，表示“True/False”。n要求覆盖“大于、小于、等于”0三种情况
1. 输出部4个new m均基于m初始值计算而得

Fork #2

### PR DESCRIPTION
**Pull request by jason-zzq**
Resolve #4

Signed-off-by: Jason Zhang <1104025131@qq.com>

### CODE REVIEW

File: operators/jason-zzq.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+# 优点：也可以在其他语言中直接执行
+# 缺点：执行效率低
+
+print(a, b)
+a = a + b; b = a - b; a = a - b
+print(a , b)
+
+a = a ^ b; b = a ^ b; a = b ^ a
+print(a, b)
```
Comment (bonnie-young) [inline operators/jason-zzq.py]: python的元组交换最方便，你没有使用哦

File: operators/jason-zzq.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
+# 优点：执行效率高
+# 缺点：在其他语言中，temp类型需要再定义
+
+a = 2; b = 3
+print(a, b)
+temp = a; a = b; b = temp
+print(a, b)
+
+# 优点：也可以在其他语言中直接执行
```
Comment (bonnie-young) [inline operators/jason-zzq.py]: 这个优点有些牵强。相比第一种方式少定义了一个变量，但缺点是需要额外的计算，执行效率低

File: operators/jason-zzq.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+# 缺点：执行效率低
+
+print(a, b)
+a = a + b; b = a - b; a = a - b
```
Comment (bonnie-young) [inline operators/jason-zzq.py]: 正数加减可能会导致溢出

File: operators/jason-zzq.py [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+print(a , b)
+
+a = a ^ b; b = a ^ b; a = b ^ a
```
Comment (bonnie-young) [inline operators/jason-zzq.py]: 异或对整数来说不会产生溢出问题，但仅限于整数，不支持浮点数

Comment (conke) [reply operators/jason-zzq.py]: 是的，或运算仅支持整数，不支持浮点型

**Comment by bonnie-young** [comment]
采用格式化输出
print("%o %d..." % (m, m1, ...))

**Comment by jason-zzq** [comment]
No. | Output line | Format | Result
----| ----------- | ------ | ------
1 | 2 3 => 3 2 | {} {} => {} {} | True
2 | 2 3 => 3 2 | {} {} => {} {} | True
3 | 2 3 => 3 2 | {} {} => {} {} | True
4 | 2 3 => 3 2 | {} {} => {} {} | True
5 | 3.1231 3.1234 0.0001 ne | {} {} {} {} | True
6 | 2080 yes | {} {} | True
7 | 24 even | {} {} | True
8 | 24 even | {} {} | True
9 | 28 no | {} {} | True
10 | 3 x 17 = 51 | {} x 17 = {} | True
11 | 17 % 8 = 1 | {} % 8 = {} | True
12 | 224 / 16 = 14 | {} / 16 = {} | True
13 | 426: 600 537 626 404 | {}: {} {} {} {} | True

**Comment by jason-zzq** [comment]
No. | Output | Format | Result
--- | ------ | ------ | ------
1 | 2 3 => 3 2 | {} {} => {} {} | True
2 | 5 6 => 6 5 | {} {} => {} {} | True
3 | 5 6 => 6 5 | {} {} => {} {} | True
4 | 5 6 => 6 5 | {} {} => {} {} | True
5 | 3.1231 3.1234 0.0001 ne | {} {} {} {} | True
6 | 2080 yes | {} {} | True
7 | 24 even | {} {} | True
8 | 24 even | {} {} | True
9 | 28 no | {} {} | True
10 | 3 x 17 = 51 | {} x 17 = {} | True
11 | 17 % 8 = 1 | {} % 8 = {} | True
12 | 224 / 16 = 14 | {} / 16 = {} | True
13 | 426: 600 537 626 404 | {}: {} {} {} {} | True

**Pull request merged**